### ISSUE TITLE
Microsoft.Orleans.TestingHost 2.0.0 support for .NetCore

### ISSUE THREAD

**Issue by pfrendo** [dotnet/orleans#3032]
The current build which is actually labeled 2.0.0 within myget-orleans-prerelease is targeting .NET4.6.2 and not .netcore

https://dotnet.myget.org/feed/orleans-prerelease/package/nuget/Microsoft.Orleans.TestingHost

**Comment by jdom** [comment]
Yes, this is a current limitation because we rely on creating AppDomains for isolating the silos, and that is not supported in .NET Standard (nor Core). Nevertheless, since we removed many of the statics (but not all yet unfortunately), we could try an approach that starts the silo directly without using AppDomains, potentially limiting to just 1 silo for now.
Regardless, the end goal is to have a .NET Core compatible TestCluster, that's for sure. Currently that feature is part of the WALK phase (and we are still in the CRAWL phase, according to https://github.com/dotnet/orleans/issues/2143), but that doesn't mean we wouldn't accept contributions for it ;)

**Comment by pms1969** [comment]
@jdom Is the preview code on master?  I might take a crack at it.  Any hints as to where to start?

**Comment by sergeybykov** [comment]
.NET Core versions of the .csproj files are under \vNext\src. We are in the process of merging them with those in \src.

**Comment by dsarfati** [comment]
@jdom @sergeybykov Is there an update on the schedule for this. It has been awhile since I have seen any updates. If there are specific tasks that need to be completed, I may have some time to help move this along so we could get single silo in-memory testing.

**Comment by sergeybykov** [comment]
@dsarfati Sorry, I missed your question over the holidays.

#3878 is a big change that should align `Orleans.TestingHost` with the new configuration API. Once that is in, we can look at what's incompatible with .NET Standard there.

**Comment by dsarfati** [comment]
@sergeybykov No worries, I completely understand. I look forward to it!

**Comment by sergeybykov** [comment]
@ReubenBond might have an up-to-date view of what's left.

**Comment by jdom** [comment]
Not much to do, we just need to make the AppDomainSiloHandle functionality into a replaceable strategy (one that just runs in the same memory space) and then multi-target the TestingSiloHost nuget package so that it targets .NET Standard (or .NET Core) also.
